### PR TITLE
Added csv reader.

### PR DESCRIPTION
**Pull request by MustyC**
# Description

Added CSV reader which reading in from .csv file. Added dependencies to pom file. Added logging to CsvReader. Created tests for CsvReader. Added test data files. Two tests are failing but will be fixed soon.

Fixes # (issue)
have two failing tests, one of which has been disabled. Will fix it shortly. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Fully Unit Tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/test/java/com/spartaglobal/moviesapi/readin/CsvReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }).isInstanceOf(IOException.class)
+        .hasMessageContaining("");
+  }
+
+  @Disabled
+  @Test
+  public void readTwoFilms() {
+    List<String[]> actualResult = CsvReader.readFile(
+        "src/test/resources/com/spartaglobal/moviesapi/test_csv_files/test_data6.csv");
+    String[] expectedResult = new String[]{"Avatar",
```
Comment (Silwhoon) [inline src/test/java/com/spartaglobal/moviesapi/readin/CsvReaderTest.java:46]: When 2 files are read, they should be stored as 2 separate string arrays. Having 2 films in 1 String array won't be useful to us. (The expectedResult must be a list of string arrays - List<String[]> - with the 2 films stored inside

Comment (KazdanP) [reply src/test/java/com/spartaglobal/moviesapi/readin/CsvReaderTest.java]: I just noticed that and it's now been fixed

**Review by Silwhoon — DISMISSED** [review]
Providing all tests run successfully this should be okay to merge

**Review by Silwhoon — APPROVED** [review]
Looks good to merge, the line coverage issue is not applicable here

**Review by Adil-Salamat — APPROVED** [review]

**Pull request merged**